rogue-thi-app/lib/backend-utils/timetable-utils.js

```diff
@@ -76,24 +76,52 @@ export async function getFriendlyTimetable (date, detailed) {
   const { timetable } = await API.getTimetable(date, detailed)
 
   return timetable
+    .flatMap(day =>
+      Object.values(day.hours)
+        .flatMap(hours => hours.map(hour => ({ date: day.date, ...hour })))
+    )
     .map(x => {
-      // parse dates
-      x.startDate = new Date(`${x.datum}T${x.von}`)
-      x.endDate = new Date(`${x.datum}T${x.bis}`)
+      // the actual timestamp consists of
+      // the date from x.date and the time from x.von
+      const startDate = new Date(x.date)
+      const brokenStartDate = new Date(x.von)
+      startDate.setHours(brokenStartDate.getHours())
+      startDate.setMinutes(brokenStartDate.getMinutes())
+      startDate.setSeconds(brokenStartDate.getSeconds())
+      const endDate = new Date(x.date)
+      const brokenEndDate = new Date(x.bis)
+      endDate.setHours(brokenEndDate.getHours())
+      endDate.setMinutes(brokenEndDate.getMinutes())
+      endDate.setSeconds(brokenEndDate.getSeconds())
 
       // normalize room order
-      if (x.raum) {
-        x.rooms = x.raum
+      let rooms
+      if (x.details.raum) {
+        rooms = x.details.raum
           .split(',')
           .map(x => x.trim().toUpperCase())
           .sort()
-        x.raum = x.rooms.join(', ')
       } else {
-        x.rooms = []
-        x.raum = ''
+        rooms = []
       }
 
-      return x
```

Comment:
I think in the original development we always agreed on just using the original data layout instead of starting to translate field names etc., since that is just additional work in converting it to english first and then understanding both formats.

Reply:
Yes, but since the API format changed and we would have to update all usages anyways I decided to convert it to a more intuitive and stable format